lease/lessor.go

```diff
@@ -160,9 +164,10 @@ func newLessor(b backend.Backend, minLeaseTTL int64) *lessor {
 		b:           b,
 		minLeaseTTL: minLeaseTTL,
 		// expiredC is a small buffered chan to avoid unnecessary blocking.
-		expiredC: make(chan []*Lease, 16),
-		stopC:    make(chan struct{}),
-		doneC:    make(chan struct{}),
+		expiredC:  make(chan []*Lease, 16),
+		stopC:     make(chan struct{}),
+		doneC:     make(chan struct{}),
+		promoting: true,
```

Comment:
`promoting` field is only used by `Promote` method. If we set it `true` and then `false` inside `Promote`,  there is no way for `Promote` to know if it has promoted before? Or if we want to randomize for every `Promote` call, we can just remove the field?